### ISSUE TITLE
Negative dimension size caused by subtracting 35 from 27 for 'max_pooling1d_12/MaxPool' (op: 'MaxPool') with input shapes: [?,27,1,128]

### ISSUE THREAD

**Issue by jeroenclaes** [rstudio/keras#110]
Dear all,

I am trying to run the following model, adapted from the following example provided by the package authors: https://rdrr.io/github/rstudio/keras/src/vignettes/examples/pretrained_word_embeddings.R.

`model <- keras_model_sequential() %>%
  layer_embedding(max_features, embedding_dims, weights=list(embeddings$matrix), input_length = maxlen, trainable=FALSE) %>%
  layer_dropout(0.2) %>%
  layer_conv_1d(filters = 128, kernel_size = 5, activation = 'relu') %>% 
  layer_max_pooling_1d(pool_size = 5) %>% 
  layer_conv_1d(filters = 128, kernel_size = 5, activation = 'relu') %>% 
  layer_max_pooling_1d(pool_size = 5) %>% 
  layer_conv_1d(filters = 128, kernel_size = 5, activation = 'relu') %>% 
  layer_max_pooling_1d(pool_size = 35) %>% 
  layer_flatten() %>% 
  layer_dense(units = 128, activation = 'relu') %>% 
  layer_dense(units = length(labels_index), activation = 'softmax')`

The model does not compile and I am seeing the following  valueError being raised:
`Negative dimension size caused by subtracting 35 from 27 for 'max_pooling1d_12/MaxPool' (op: 'MaxPool') with input shapes: [?,27,1,128]`
 
I have done some searching and I think it may be related to the following Keras (the python package, that is) issue:
https://github.com/fchollet/keras/issues/3945

The solution being provided there is to set the `dim_ordering` parameter appropriately (i.e., in this case to indicate a tensorflow ordering of the dimensions).  Is there some (hidden) option to do this in R Keras?

Thanks!

**Comment by jjallaire** [comment]
The new name for that parameter is `data_format` and the R interface does include it: https://rstudio.github.io/keras/reference/layer_max_pooling_2d.html. The values are "channels_first" and "channels_last" (TensorFlow uses "channels_last")

**Comment by jeroenclaes** [comment]
Thanks, I missed that parameter!
